blocks/editor-colors/index.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * External dependencies
+ */
+import { isEmpty } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { ifCondition, withContext } from '@wordpress/components';
+import { compose } from '@wordpress/element';
+
+export default compose(
+	withContext( 'editor' )(
+		( settings, props ) => ( {
+			colors: props.colors || settings.colors,
+			disableCustomColors: props.disableCustomColors !== undefined ?
+				props.disableCustomColors :
+				settings.disableCustomColors,
+		} )
+	),
+	ifCondition(
+		( { colors, disableCustomColors } ) => ! ( isEmpty( colors ) && disableCustomColors )
```

Comment:
I spent far too much time staring at this to make sense of it. Seems correct, not sure how we could make it clearer, but doing [De Morgan's](https://en.wikipedia.org/wiki/De_Morgan%27s_laws) in my head a few times and thinking in inverses with disabling made this quite difficult to discern.